### ISSUE TITLE
Case-sensitive exact match should trump all

### ISSUE THREAD

**Issue by clauderic** [kentcdodds/match-sorter#20]
Currently, because of the way the values are toLowerCase-ed before comparison (https://github.com/kentcdodds/match-sorter/blob/master/src/index.js#L203), equal lowercased values (`bob`, `Bob`, `boB`) will be returned in their original order, with no priority given to a case-sensitive exact match. Here's an example to illustrate

```javascript
const list = ['bob', 'Bob', 'Bobby', 'Janice'];
matchSorter(list, 'Bob') // will return ['bob', 'Bob', 'Bobby']
```

Presumably the first result returned should instead be the exact case-sensitive match (`Bob`).

One way to solve this would be to add a new CASE_SENSITIVE_EQUAL threshold. Thoughts?

**Comment by kentcdodds** [comment]
Yes, this sounds perfect. Thanks!

**Comment by clauderic** [comment]
This was implemented and merged with #22 🎉
